docs/src/research/installation.md

```diff
@@ -0,0 +1,71 @@
+<!--
+    Copyright 2022-2023 TII (SSRC) and the Ghaf contributors
+    SPDX-License-Identifier: CC-BY-SA-4.0
+-->
+
+# Approaches to Ghaf System Installation
+
+A hardened system installation covers multiple phases - from establishing trust to installation process. This document discusses learnings developing mechanisms to setup a Ghaf system in a target hardware.
+
+## Approaches
+
+### Ghaf Initial Approach
+
+The initial Ghaf installation approach to using Ghaf in development and demos is to build target system raw images (`img`) as binary disk images. The process results in image based on modular and configurable declarations that are repeatably built using NixOS tooling.
+
+In practice, Ghaf disk images are built with:
+
+```
+nix build .#package.<hardware-architecture>.<target-device-[release|debug]>
+```
+which results in disk image:
+```
+result\nixos.img
+```
+For information on how to build and run a Ghaf image, see [Build & Run](https://tiiuae.github.io/ghaf/ref_impl/build_and_run.html) for details.
+
+The initial Ghaf installation approach differed from NixOS installation approach:
+
+ * The key reason in Ghaf was practical: initially, it is simple to write a specific target disk image to a USB boot media or target HW internal persistent media.
+ * The NixOS approach is more generic: supporting as many devices as possible. Similar to other Linux distributions like Ubuntu or Fedora.
+
+The development objective of Ghaf is to support a portable secure system that results in a target device-specific small trusted computing base. In practice, this means that Ghaf installations are by design not meant to support a generic Linux kernel with about all the device drivers (modules) out there like Ubuntu or Fedora. Ghaf reference installations are designed and to be developed to support very specific (declaratively) hardened host and guest kernels with limited drivers only. Ghaf approach significantly reduces the size of the trusted computing base as the unneeded modules and kernel parts are not taken into use.
+
+### NixOS Approach
+
+[NixOS installation](https://nixos.org/manual/nixos/stable/#ch-installation) is well documented thus only summarized here. They key in the NixOS approach is to have a generic, bootable installation media (`iso`) like any other Linux distribution. As the NixOS installer aims to support as many devices as possible: he installer has a generic kernel (per hardware architecture), hardware recognition script, and generic requirements for system partitioning (`boot` and `root` partitions).
```

Comment:
(nit) typo: "he installer"

Reply:
Fixed in [5473018](https://github.com/tiiuae/ghaf/commit/54730185792e5264cd473843637dcab7505f6170)